### DISCUSSION TITLE
Tutorial to get the relative permeability curves

### DISCUSSION THREAD

**Discussion by minzhangut** [PMEAL/OpenPNM#2351] [Q&A]
I am wondering whether there is a tutorial that I can refer to simulate two-phase flow in the pore network and obtain the relative permeability curves. Thanks!

**Comment by jgostick** [comment]
There are two here:  https://github.com/PMEAL/OpenPNM/tree/dev/examples/simulations/percolation

One shows how to use the `metrics` class, which tries to do it for you.  The other one does relative diffusivity but the idea is the same, use StokesFlow instead of FickianDiffusion.

**Reply by minzhangut** [reply]
I tried to follow this example: https://github.com/PMEAL/OpenPNM/blob/dev/examples/simulations/percolation/relative_permeability_metrics.ipynb
I installed the dev versions of both porespy and openpnm.

When I tried to run:
`rp = op.algorithms.metrics.RelativePermeability(network=pn)`

I got the error of complaining `AttributeError: module 'openpnm.algorithms' has no attribute 'metrics'`

So I am wondering which algorithm I could use to calculate the relative permeability curves?

**Reply by Fioti38** [reply]
> I got the error of complaining `AttributeError: module 'openpnm.algorithms' has no attribute 'metrics'`

This means that you didn't install the last version of openpnm, install the dev version directly from git so that you will get the full package

**Reply by minzhangut** [reply]
I just installed the latest dev version of openpnm using "pip install git+https://github.com/PMEAL/OpenPNM.git@dev". But I still got the same error. Also, from the Module Reference, I did not see 'metrics' in 'openpnm.algorithms'. From the Module Reference, I saw 'openpnm.metrics', but I did not see "RelativePermeability" there.

**Reply by Fioti38** [reply]
![image](https://user-images.githubusercontent.com/104492138/165589123-60909701-42d8-47b0-84e6-d0e468a40e13.png)
This the version I have 2.8.2 , and there is Relativepermeability.py

**Reply by minzhangut** [reply]
I tried openpnm 2.8.2, I had a different error (and that's why I was directed to the installation of the openpnm dev version):
```
------------------------------------------------------------
ERROR      : throat.hydraulic_conductance was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-04-27 12:58:03,084
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.hydraulic_conductance was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-04-27 12:58:03,156
------------------------------------------------------------
```
